src/Query/MultiMatchQuery.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+
+namespace Novaway\ElasticsearchClient\Query;
+
+
+class MultiMatchQuery implements Query
+{
+    const BEST_FIELDS = 'best_fields';
+    const MOST_FIELDS = 'most_fields';
+    const CROSS_FIELDS = 'cross_fields';
+    const PHRASE = 'phrase';
+    const PHRASE_PREFIX = 'phrase_prefix';
+    /** @var string */
+    private $value;
+    /** @var string[]|BoostableField[] */
+    private $fields;
+    /** @var array */
+    private $options;
+    /** @var string */
+    private $combiningFactor;
+
+    /**
+     * @param string $value the query value to search for
+     * @param array $fields the fields to search for. Should either contain strings or BoostableField
+     * @param string $combiningFactor the combining factor
+     * @param array $options additional options
+     */
+    public function __construct(string $value, array $fields, string $combiningFactor = CombiningFactor::SHOULD,array $options = [])
+    {
+        array_map(function($field) {
```

Comment:
`array_walk` might be a better choice since no returned array is expected.
https://secure.php.net/manual/en/function.array-walk.php